### DISCUSSION TITLE
PyQtGraph maintainer here, I see you're using ImageItem, some elements to keep in mind...

### DISCUSSION THREAD

**Discussion by j9ac9k** [EnMAP-Box/enmap-box#905] [General]
Hello folks,

I maintain pyqtgraph, I see you use our ImageItem grpahics object, which is great.  You may or may not be aware, but ImgaeItem's performance, can _rapidly_ change depending on some of the attributes that are being passed in (dtypes, size of the lookup table, if you're using levels, the shape of the levels, etc etc).  It actually got confusing enough that I was no longer aware which usages of ImageItem were "fast" and "slow", so I crafted a PR (about to be merged, but not yet) that tries to lay things out a bit better for what to try and what not do for performance:

https://pyqtgraph--3028.org.readthedocs.build/en/3028/api_reference/graphicsItems/imageitem.html#performance

Taking a quick peek at your codebase, I see you already has lookup tables of dtype uint8 and no longer than 256 elements:
https://github.com/EnMAP-Box/enmap-box/blob/676d779a9c1979b4d34461f098c1ca9e4b8ef7ba/enmapbox/coreapps/scatterplotapp/scatterplotdialog.py#L492-L502

Also looks like you're plotting data that is of type ~~uint8~~ float, the only other suggestion I have is make sure you specify ~~`levels=None`~~ (EDIT: ``levels=None`` does not work with float dtype) levels limit to avoid auto-sampling. 

Since you have float dtypes, ...how is it not raising an issue w/o you specifying levels... 

Anyway if there are other performance bottlenecks or questions you have regarding pyqtgraph, feel free to reach out.

**Comment by janzandr** [comment]
Thanks for reaching out. And many thanks for the great work you did with PyQtGraph over the last years... it really helped us a lot.

**Reply by j9ac9k** [reply]
I'm glad to hear the library has worked well for you all!  Let me know if there are any issues or questions.

I did give you one bad bit of bogus advice about ``levels=None``, that doesn't work w/ float dtypes.  Sorry about that!
